config/confighttp/confighttp.go

```diff
@@ -57,7 +57,7 @@ type ClientConfig struct {
 	Headers map[string]configopaque.String `mapstructure:"headers"`
 
 	// Custom Round Tripper to allow for individual components to intercept HTTP requests
-	CustomRoundTripper func(next http.RoundTripper) (http.RoundTripper, error)
+	CustomRoundTripper func(next http.RoundTripper) (http.RoundTripper, error) `mapstructure:"-"`
```

Comment:
Directly calling `yaml.Marshal` on `ClientConfig` will still fail, this only helps when we go through confmap.